scraping/lapasta-scraper.js

```diff
@@ -0,0 +1,93 @@
+"use strict";
+
+const Promise = require('bluebird');
+const request = Promise.promisifyAll(require("request"));
+const cheerio = require('cheerio');
+const moment = require('moment');
+const Food = require("../models/food");
+const Menu = require("../models/menu");
+const config = require('../config');
+const timeHelper = require('../helpers/timeHelper');
+const scraperHelper = require('./scraperHelper')
+
+var LapastaUrl = config.scraper.lapastaUrl;
+
+function _getWeekPlan() {
+  return request.getAsync(LapastaUrl)
+          .then(res => res.body)
+          .then(body => _parseHTML(body));
+}
+
+function _parseHTML(html) {
+  var weekPlan = new Array(7);
+
+  // Lapasta is closed on Saturday and Sunday
+  let closedMenu = new Menu();
+  closedMenu.closed = true;
+  weekPlan[5] = closedMenu;
+  weekPlan[6] = closedMenu;
+
+  // Init parser
+  var $ = cheerio.load(html);
+
+  // Load day-divs
+  var dayDivs = $('#menue-container').find('.ce_text.block');
+
+  // Get monday div and compare date to current date
+  let mondayDate = moment(dayDivs.first().find("h2").text().split(", ")[1], "DD.MM.");
+  if (mondayDate.isValid() && mondayDate.format("D.M") !== timeHelper.getMondayDate()) {
+      return scraperHelper.invalidateMenus(weekPlan);
+  }
+
+  // weekdays
+  for (let dayInWeek = 0; dayInWeek < 5; dayInWeek++) {
+    var dayEntry = dayDivs.eq(dayInWeek);
+    try {
+        weekPlan[dayInWeek] = _parseDayMenu(dayEntry);
+    } catch (ex) {
+        console.log("error", ex)
+        let errorMenu = new Menu();
+        errorMenu.error = true;
+        weekPlan[dayInWeek] = errorMenu;
+    }
+  }
+  return weekPlan;
+}
+
+function _parseDayMenu(dayEntry) {
```

Comment:
Vielleicht lerne ich noch was: gibts einen Grund für hier Underscore-Methodennamen?

Reply:
Underscore wird oft als Kennzeichner für private Methoden verwendet. Ich bin vor allem in Sprachen wie JS ein Fan davon weil man dadurch gleich weiß dass etwas nicht public ist.

Reply:
https://stackoverflow.com/a/4484449/2428434

Reply:
Gut, ja, das weiß ich. Man kann natürlich diskutieren, wie private eine Modulmethode ist, wenn du sie dann exportierst und in einem anderen Modul verwendest, aber mir ist das eigentlich zu blöd. 

OT: ich finde, man sollte denn Scraper insgesamt stärker fragmentieren; jetzt ist das File schon sehr unübersichtlich, da gefällt mir deine Lösung gut.